site/src/pages/GroupsPage/GroupPage.tsx

```diff
@@ -151,26 +112,32 @@ export const GroupPage: React.FC = () => {
               <Maybe
                 condition={
                   canUpdateGroup &&
-                  group !== undefined &&
-                  !isEveryoneGroup(group)
+                  groupData !== undefined &&
+                  !isEveryoneGroup(groupData)
                 }
               >
                 <AddGroupMember
-                  isLoading={state.matches("addingMember")}
-                  onSubmit={(user, reset) => {
-                    send({
-                      type: "ADD_MEMBER",
-                      userId: user.id,
-                      callback: reset,
-                    });
+                  isLoading={addMemberMutation.isLoading}
+                  onSubmit={async (user, reset) => {
```

Comment:
Just making sure I'm understanding this API: the reason the `reset` function is exposed to the parent is because the function shouldn't be called unconditionally. And if we had tried keeping the `reset` implementation detail private to `AddGroupMember`, every `onSubmit` callback would need to re-throw its error so that `AddGroupMember` can catch it locally and choose not to reset.

Is that basically it?

Reply:
The `reset` is used to reset the form - which is used by its children - after the submission. I didn't find in the past a better way to do that 🤔 but I'm open to suggestions.

Reply:
No, the use case makes sense to me – I guess I just haven't happened to run into a component that needed this level of control before, so I was making sure I was reading this properly